src/components/Trade/TradeInput.tsx

```diff
@@ -102,14 +152,16 @@ export const TradeInput = ({ history }: RouterProps) => {
           <FormErrorMessage>{errors.fiatAmount && errors.fiatAmount.message}</FormErrorMessage>
         </FormControl>
         <FormControl>
-          <TokenRow
+          <TokenRow<TradeState<ChainTypes, SwapperType>>
             control={control}
             fieldName='sellAsset.amount'
             rules={{ required: true }}
-            onInputChange={(value: string) => {
-              const action = value ? TradeActions.SELL : undefined
-              action ? setValue('action', action) : reset()
-              getQuote({ sellAmount: value }, sellAsset, buyAsset)
+            onInputChange={(amount: string) => {
+              if (Number(amount) !== Number(sellAsset.amount)) {
```

Comment:
I assume `amount` and `sellAsset.amount` are strings? I'd prefer to use BigNumber to compare as it won't have precision errors.